packages/transport/src/modules/relationships/RelationshipsController.ts

```diff
@@ -197,7 +196,49 @@ export class RelationshipsController extends TransportController {
 
         this.eventBus.publish(new RelationshipChangedEvent(this.parent.identity.address.toString(), newRelationship));
 
-        return newRelationship;
+        return Result.ok(newRelationship);
+    }
+
+    public async canSendRelationship(parameters: ISendRelationshipParameters): Promise<Result<void>> {
+        const template = (parameters as SendRelationshipParameters).template;
+        if (!template.cache) {
+            throw this.newCacheEmptyError(RelationshipTemplate, template.id.toString());
+        }
+
+        const peerAddress = template.cache.createdBy;
+        const existingRelationshipToPeer = await this.getExistingRelationshipToIdentity(peerAddress);
+
+        if (existingRelationshipToPeer) {
+            return Result.fail(TransportCoreErrors.relationships.relationshipCurrentlyExists(existingRelationshipToPeer.status));
+        }
+
+        if (template.isExpired()) {
+            return Result.fail(TransportCoreErrors.relationships.relationshipTemplateIsExpired(template.id.toString()));
+        }
+
+        const result = await this.client.canCreateRelationship(peerAddress.toString());
+
+        if (!result.value.canCreate) {
+            return Result.fail(
+                new ApplicationError(
+                    "error.transport.relationships.relationshipNotYetDecomposedByPeerOrPeerIsToBeDeleted",
+                    `A Relationship to the peer ${peerAddress} cannot be created, because the former Relationship is not yet decomposed by the peer or the peer is to be deleted.`
```

Comment:
I don't think that the canCreate check should provide a less detailed result than the Create check - realistically the detailed Create results will never be seen

Reply:
Also, if I remember correctly, it was decided in the meeting that the detailed error should be thrown in both cases. Wasn't it so?